### ISSUE TITLE
Compilation failures in non-Cuda builds using nvcc_wrapper

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#5665]
**Describe the bug**

Compiling Kokkos using nvcc_wrapper without Cuda result in compilation errors as below:

```
23:55:38 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkos-dev-2_nvccwrapper_nocuda_gcc830/kokkos/core/src/../../tpls/desul/include/desul/atomics/Fetch_Op_ScopeCaller.hpp(45): error: calling a __host__ function("Test::long_wrapper::operator =") from a __device__ function("desul::Impl::device_atomic_fetch_oper< ::desul::Impl::store_operator< ::Test::long_wrapper, const  ::Test::long_wrapper> ,  ::Test::long_wrapper,  ::desul::MemoryOrderRelease> ") is not allowed
```

**Please also include the following items to support reproducing the bug**
**Reproducer** (kokkos-dev-2):
```bash

source /projects/sems/modulefiles/utils/sems-archive-modules-init.sh
module purge
module load sems-archive-env
module load sems-archive-cuda/11.1 sems-cmake/3.18.4 sems-archive-gcc/8.3.0

cmake -DCMAKE_CXX_COMPILER=$KOKKOS_PATH/bin/nvcc_wrapper -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF $KOKKOS_PATH
```

**Comment by ajpowelsnl** [comment]
@ndellingwood - did we ever promise support for this (i.e., using the wrapper without CUDA)?

**Comment by ndellingwood** [comment]
@ajpowelsnl not sure if the support is promised, but it had occurred previously with #5093 which was resolved by #5115 and was the motivation to set up a nightly build

**Comment by ajpowelsnl** [comment]
> @ajpowelsnl not sure if the support is promised, but it had occurred previously with #5093 which was resolved by #5115 and was the motivation to set up a nightly build

Hmmm ... good point.  Let me look into this a bit further.

**Comment by dalg24** [comment]
I looked into this and it is a Desul issue that I had brought up and that we briefly discussed with @trws and @crtrott but we somehow forgot about it.
Essentially Desul detects that it is being compiled with NVCC and does not have a configure-time option to say "disable CUDA".
